impl/client/src/main/javascript/web/util/xss.js

```diff
@@ -142,8 +142,27 @@ define("common-ui/util/xss", ["common-ui/dompurify"], function(DOMPurify) {
      */
     setHtmlUnsafe: function(elem, unsafeHtml) {
       elem.innerHTML = unsafeHtml;
-    }
-  });
+    },
+
+    /**
+     * Set's an anchor HTML element to a given sanitized Url.
+     *
+     * @param url
+     */
+    sanitizeUrl: function(url) {
```

Comment:
I wonder why the `DOMPurify` library didn't use a similar implementation, instead of using their own reg exp? 

Could they want to support any non-strict URL formats which might be common out there, that `href` attributes accept, but not `new URL(.)`?

To support relative URLs, a use case which may be relevant to us, you should do instead `new URL(url, document.location.href)`.

Not sure if any other protocols would be needed. It doesn't seem so.

Otherwise, this seems enough for our purposes.

I would

Reply:
Despite my considerations, this was my only direct ask:
> To support relative URLs, a use case which may be relevant to us, you should do instead `new URL(url, document.location.href)`.

Instead, you're going back to regular expressions, and still did not address the relative URL case.
I would actually prefer the safety and "always up to date"'ness of the standard URL implementation. Otherwise, if you think the direct regular expression has advantages, please reply with which. Maybe even write those as a comment in the code.